### PR TITLE
Task-47896: Fix edit project - refresh issues

### PR DESCRIPTION
**Pull request by NaderSai**
Problem: when changing the project from Project Tab it doesn't update in the board/list view. 
How it was solved: by sending event when updating the project to the board event to update it directly in the board view. And by affecting the false value to isPersonnalTask to select the right div for the list view.

### CODE REVIEW

File: task-management/src/main/webapp/vue-app/tasks-management/components/tasks/TasksListItem.vue [hunk 1]
```diff
@@ -214,6 +214,11 @@ export default {
       this.updateTaskCoworker(value,id);
     });
     this.getTaskAssigneeAndCoworkers();
+    this.$root.$on('update-task-widget-list', task => {
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/tasks-management/components/tasks/TasksListItem.vue]: Can you explain this part ? I'm not sure to understand.

Comment (NaderSai) [reply task-management/src/main/webapp/vue-app/tasks-management/components/tasks/TasksListItem.vue]: at the top (line 42) there is a condition in the div in which we find the project name implementation. According to the written code we can enter to this div only if isPersonnalTask value is false. so when we receive the event update-task-widget-list which is related to the updateTask method we must put isPersonnalTask to false to consider the update directly without need to refresh the page.

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you rebase your pr to have only your commit ?

**Comment by NaderSai** [comment]
> Can you rebase your pr to have only your commit ?

done !

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskProject.vue [hunk 2]
```diff
@@ -148,6 +148,12 @@ export default {
         this.task.status = status;
         this.updateTask();
         this.projectModel = project;
+      }).then( () => {
+        this.$root.$emit('update-task-project', this.task)
+          .catch(e => {
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskProject.vue]: this catch generate a js error :
I create 2 spaces
I add a task in the space project
I reopen the task, and choose another project, and then 
image

File: task-management/src/main/webapp/vue-app/tasks-management/components/tasks/TasksListItem.vue [hunk 3]
```diff
@@ -214,6 +214,11 @@ export default {
       this.updateTaskCoworker(value,id);
     });
     this.getTaskAssigneeAndCoworkers();
+    this.$root.$on('update-task-project', task => {
+      if (this.task.task.status && this.task.task.id===task.id){
+        this.isPersonnalTask=false;
+      }
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/tasks-management/components/tasks/TasksListItem.vue:220]: This code is not called when updating a task project. I think it should be removed

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you rebase the PR to remove other commits ?

File: task-management/src/main/webapp/vue-app/tasks-management/components/tasks/TasksListItem.vue (hunk 3 above)
Comment (NaderSai) [reply task-management/src/main/webapp/vue-app/tasks-management/components/tasks/TasksListItem.vue:220]: > Can you rebase the PR to remove other commits ?

done

**Review by rdenarie — CHANGES_REQUESTED** [review]
One thing is missing, can you put the PR description in the first commit message ?
Thanks

**Review by rdenarie — APPROVED** [review]

**Pull request merged**